File: Valid-Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st=[]
        n=len(s)

        for i in range(n):
            if(s[i]=='('or s[i]=='[' or s[i]=='{'):
                st.append(s[i])
            else:
                if(len(st)==0):
                    return False
                ch=st[-1]

                if (ch=='('and s[i]==')') or (ch=='['and s[i]==']') or (ch=='{'and s[i]=='}'):
                    st.pop()
                else:
                    return False
        if len(st)==0:
            return True
        else:
            return False

File: Valid-Parentheses.cpp
1class Solution {
2public:
3    bool isValid(string s) {
4        int n=s.size();
5        stack<char>st;
6        for(int i=0;i<n;i++){
7            char ch=s[i];
8            if(s[i]=='('||s[i]=='{'||s[i]=='['){
9                st.push(s[i]);
10            }
11            else{
12                if(st.empty()){
13                    return false;
14                }
15                char top=st.top();
16                if(top=='('&&s[i]==')'||top=='{'&&s[i]=='}'||top=='['&&s[i]==']'){
17                    st.pop();
18                }
19                else{
20                    return false;
21                }
22            }
23        }
24        if(st.empty()){
25            return true;
26        }
27        else return false;
28    }
29};